app/views/transactions/destroy.js.erb

```diff
@@ -1,4 +1,14 @@
-$('.transactions_list').html("<%=j render partial: 'transactions/transactions_table', locals: { transactions: @transactions } %>");
+$("#<%= dom_id(@transaction) %>").remove();
+$("#<%= dom_id(@transaction, 'edit_row') %>").remove();
+
+if ($('.transactions tbody tr').size() == 0) {
+  if (document.URL != "<%= root_url %>") {
```

Comment:
@Gataniel Instead checking url just check if search form already exists. And just pass this condition as parameter